code/game/machinery/_machinery.dm

```diff
@@ -146,6 +148,18 @@ Class Procs:
 
 	return INITIALIZE_HINT_LATELOAD
 
+/obj/machinery/proc/get_occupant()
```

Comment:
this should not exist at all

Reply:
To expand on this, while this is a good pattern on languages that let you encapsulate variables into private permissions, that feature is not present for DM and it is rather unenforceable.

This combined with the non-trivial cost of proc overhead, compared with more sane languages, make getter procs a bad choice without a good reason here.

There's another reason that makes me want to avoid it, and that is the temptation of coders to redefine this proc to add snowflake checks that should be handled elsewhere, but that's besides the point.